### PR TITLE
Migrate composition of the overlay and pointer to the present thread

### PR DESCRIPTION
**Pull request by jherico**

### CODE REVIEW

**Comment by jherico** [comment]
https://app.asana.com/0/27650181942747/92628920509312

File: interface/src/ui/ApplicationCompositor.cpp [hunk 1]
```diff
@@ -1,690 +0,0 @@
...
-}
-
-bool ApplicationCompositor::getReticleOverDesktop() const { 
-    // if the QML/Offscreen UI thinks we're over the desktop, then we are...
-    // but... if we're outside of the overlay area, we also want to call ourselves
-    // as being over the desktop.
-    if (qApp->isHMDMode()) {
-        QMutexLocker locker(&_reticleLock);
-        glm::vec2 maxOverlayPosition = qApp->getUiSize();
-        if (_reticlePositionInHMD.x < 0 || _reticlePositionInHMD.y < 0 ||
```
Comment (ZappoMan) [inline interface/src/ui/ApplicationCompositor.cpp]: you appear to have removed this logic... was that intentional? Why?

the _isOverDesktop will only be set if you're actually over the Desktop.qml... but we need to also consider all the space outside of the Desktop to count as "desktop" as opposed to a window/overlay....

**Comment by hifi-gustavo** [comment]
Build console output for commit: e6e7708b5e47a4d4246ee96902024b7a455c81f6 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 52970fe80f790a3e7dd37a116d02a5bd0e09c3fd available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

File: interface/src/ui/ApplicationCompositor.cpp (hunk 1 above)
Comment (jherico) [inline interface/src/ui/ApplicationCompositor.cpp]: will fix

**Comment by hifi-gustavo** [comment]
Build console output for commit: b325b1efefe261c5efca44eca548fad86dca70c6 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

File: interface/src/ui/ApplicationCompositor.cpp (hunk 1 above)
Comment (jherico) [inline interface/src/ui/ApplicationCompositor.cpp]: fixed

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+    return _currentDisplayPlugin && _currentDisplayPlugin->isHmd();
+}
+
+bool CompositorHelper::shouldCaptureMouse() const {
+    // if we're in HMD mode, and some window of ours is active, but we're not currently showing a popup menu
+    return isHMD() && QApplication::activeWindow(); // && !Menu::isSomeSubmenuShown();
+}
+
+void CompositorHelper::handleLeaveEvent() {
+#if 0
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: we need this behavior as well... is there a reason you if 0'd this out?

**Comment by hifi-gustavo** [comment]
Build console output for commit: d6c65fcf30bb6e50c6d153f368518c8e3d786ec5 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 467b2b3278528aeadae2c1eb90b3c6d352ae1965 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 69d6b4c051c0441cd05b3bfc60680875ef5ef7a1 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d69c6a7b8d036a72e77a7a9a6db890486304d188 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b32e5dbf108e7474326cc53c9b03fa431020d094 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0f0bcfb02ce9388b5d8dee935f38b4e9b65be922 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: eaf6e5516e4a2d80eafab229c36607478ba8d244 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 135ecc57d56276e65228f1513a40d774e9cce749 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: fb07425d0d83c8f13f82106b7d4a145aa4a219e1 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 680e660a827e7e74f1d426f2d80408e3f8f34ff3 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 522976411f28b0b82d6bcced6c0abbdc1f155a67 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a456e21564914499e2e0dffe95631f59c2146efd available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

File: interface/src/ui/ApplicationOverlay.cpp [hunk 3]
```diff
@@ -244,34 +244,69 @@ void ApplicationOverlay::renderDomainConnectionStatusBorder(RenderArgs* renderAr
...
+std::mutex _textureGuard;
+using Lock = std::unique_lock<std::mutex>;
+std::queue<gpu::TexturePointer> _availableTextures;
+
 void ApplicationOverlay::buildFramebufferObject() {
     PROFILE_RANGE(__FUNCTION__);
 
     auto uiSize = qApp->getUiSize();
+    if (!_overlayFramebuffer || uiSize != _overlayFramebuffer->getSize()) {
+        _overlayFramebuffer = gpu::FramebufferPointer(gpu::Framebuffer::create());
```
Comment (jherico) [inline interface/src/ui/ApplicationOverlay.cpp]: originally I tried to re-use the existing overlay, but managed to get into some weird state where the depth and color buffers ended up the wrong size.  This approach is simpler.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3227b4594b1a706dc4bc319431b95db783c77523 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

File: libraries/gl/src/gl/GLEscrow.h [hunk 4]
```diff
@@ -42,13 +42,19 @@
...
+    //,
+    //// Only accept numeric types
+    //typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type
 >
 class GLEscrow {
 public:
     static const uint64_t MAX_UNSIGNALED_TIME = USECS_PER_SECOND / 2;
+    
+    const T& invalid() const {
+        static const T INVALID_RESULT;
```
Comment (jherico) [inline libraries/gl/src/gl/GLEscrow.h]: This works if T is a class type with a constructor, but not if T is a numeric type.  Hence, there is a specialization below specifically for `GLuint` that initializes it to 0.

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 5]
```diff
@@ -0,0 +1,483 @@
...
+        *result = t0;
+        return true;
+    }
+}
+
+CompositorHelper::CompositorHelper() :
+    _alphaPropertyAnimation(new QPropertyAnimation(this, "alpha")),
+    _reticleInterface(new ReticleInterface(this))
+{
+    // FIX in a separate PR addressing the current mouse over entity bug
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: Is this only for the link and market place support? This really should be in the EntityTreeRenderer instead I think...

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 6]
```diff
@@ -0,0 +1,483 @@
...
+    if (shouldCaptureMouse()) {
+        
+        //QWidget* mainWidget = (QWidget*)qApp->getWindow();
+        static auto renderingWidget = PluginContainer::getInstance().getPrimaryWidget();
+        static QWidget* mainWidget = nullptr;
+        if (mainWidget == nullptr) {
+            mainWidget = renderingWidget->parentWidget();
+        }
+        QRect mainWidgetFrame;
+        {
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: do we really need this extra scope here?

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 7]
```diff
@@ -0,0 +1,483 @@
...
+        result = glm::vec2(rec.right(), rec.bottom());
+    }
+    return result;
+}
+
+void CompositorHelper::setReticlePosition(const glm::vec2& position, bool sendFakeEvent) {
+    if (isHMD()) {
+        QMutexLocker locker(&_reticleLock);
+        glm::vec2 maxOverlayPosition = _currentDisplayPlugin->getRecommendedUiSize();
+        // FIXME don't allow negative mouseExtra
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: incorrect -- negative mouse extra is good and needed, so that we can have the mouse reticle go above the Overlay portion and intersect with 3D objects... **please remove this FIXME comment**

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 8]
```diff
@@ -0,0 +1,483 @@
...
+        _reticlePositionInHMD = glm::clamp(position, minMouse, maxMouse);
+
+        if (sendFakeEvent) {
+            // in HMD mode we need to fake our mouse moves...
+            QPoint globalPos(_reticlePositionInHMD.x, _reticlePositionInHMD.y);
+            auto button = Qt::NoButton;
+            auto buttons = QApplication::mouseButtons();
+            auto modifiers = QApplication::keyboardModifiers();
+            if (qApp->thread() == QThread::currentThread()) {
+                QMouseEvent event(QEvent::MouseMove, globalPos, button, buttons, modifiers);
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: how did you make sure that this event is properly interpreted as a "fakeMouse" and not a real mouse event?

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 9]
```diff
@@ -0,0 +1,483 @@
...
+            mat4 cameraMat = _currentCamera;
+            auto UITransform = cameraMat * glm::inverse(eyePose);
+            auto relativePosition4 = glm::inverse(UITransform) * vec4(apparentPosition, 1);
+            auto relativePosition = vec3(relativePosition4) / relativePosition4.w;
+            auto relativeDistance = glm::length(relativePosition);
+
+            // look at borrowed from overlays
+            float elevation = -asinf(relativePosition.y / glm::length(relativePosition));
+            float azimuth = atan2f(relativePosition.x, relativePosition.z);
+            glm::quat faceCamera = glm::quat(glm::vec3(elevation, azimuth, 0)) * quat(vec3(0, 0, -1)); // this extra *quat(vec3(0,0,-1)) was required to get the quad to flip this seems like we could optimize
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: this is wrong you missed a bug fix

should be:
```
 glm::quat faceCamera = glm::quat(glm::vec3(elevation, azimuth, 0)) * quat(vec3(0, -PI, 0));
```

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 10]
```diff
@@ -0,0 +1,483 @@
...
+        auto reticlePosition = getReticlePosition();
+        if (getReticleDepth() != 1.0f) {
+            // calculate the "apparent location" based on the depth and the current ray
+            glm::vec3 origin, direction;
+            computeHmdPickRay(reticlePosition, origin, direction);
+            auto apparentPosition = origin + (direction * getReticleDepth());
+
+            // same code as used to render for apparent location
+            mat4 cameraMat = _currentCamera;
+            auto UITransform = cameraMat * glm::inverse(eyePose);
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: shouldn't you use your fancy new ` getUiTransform();`?

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 11]
```diff
@@ -0,0 +1,483 @@
...
+
+            // look at borrowed from overlays
+            float elevation = -asinf(relativePosition.y / glm::length(relativePosition));
+            float azimuth = atan2f(relativePosition.x, relativePosition.z);
+            glm::quat faceCamera = glm::quat(glm::vec3(elevation, azimuth, 0)) * quat(vec3(0, 0, -1)); // this extra *quat(vec3(0,0,-1)) was required to get the quad to flip this seems like we could optimize
+
+            Transform transform;
+            transform.setTranslation(relativePosition);
+            transform.setScale(reticleScale);
+            transform.postScale(relativeDistance); // scale not quite working, distant things too large
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: note: this comment is wrong... I know it's in production, but I've sense confirmed that the scale is working properly

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 12]
```diff
@@ -0,0 +1,483 @@
...
+
+void CompositorHelper::toggle() {
+    if (_fadeInAlpha) {
+        fadeOut();
+    } else {
+        fadeIn();
+    }
+}
+
+glm::mat4 CompositorHelper::getReticleTransform(const glm::mat4& eyePose) const {
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: you appear to have accidentally removed support for `getReticleVisible()`... or I'm missing where you're checking that in the plugins.

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp (hunk 10 above)
Comment (jherico) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: no, because that head pose is done with the 'current head pose' while this one will be done specifically with the exact head pose used to render the scene.  If I used the other, it would cause judder in the reticle as you moved your head.

Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: hmmm - ok - I will test.

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp (hunk 11 above)
Comment (jherico) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: will fix

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp (hunk 9 above)
Comment (jherico) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: will fix

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp (hunk 7 above)
Comment (jherico) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: my concern here is what if the overlay is larger than the mouseExtra area.

**Comment by ZappoMan** [comment]
My initial smoke tests shows a bunch of problems:
- depth reticle is not working
- playa has really really bad performance, I was getting a present rate of 11-17...
- deadlocked on startup the first time (likely because I was starting in HMD mode?)

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp (hunk 10 above)
Comment (jherico) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: basically, for scripts, which are asynchronous, the 'latest head pose' is typically sufficient.  However, for rendering, we need to match the exact head pose that will be used to render the overlay and which was used to render the scene.

**Comment by jherico** [comment]
Looks like the depth reticle is only working when you're completely off the overlay, still investigating.
Can't reproduce bad playa performance or deadlocking.

File: libraries/ui/src/ui/Menu.cpp [hunk 13]
```diff
@@ -0,0 +1,611 @@
...
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "Menu.h"
+
+#include <QtCore/QThread>
+#include <QtCore/QDebug>
+#include <QtWidgets/QShortcut>
+
+//#include <AddressManager.h>
```
Comment (ZappoMan) [inline libraries/ui/src/ui/Menu.cpp]: dead code

File: interface/src/Menu.h [hunk 14]
```diff
@@ -9,144 +9,20 @@
 //  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
 //
 
+#include <ui/Menu.h>
+
 #ifndef hifi_Menu_h
```
Comment (ZappoMan) [inline interface/src/Menu.h]: according the coding standard this should be "AppMenu.h"

File: interface/src/Menu.cpp [hunk 15]
```diff
@@ -45,14 +46,14 @@
 
 #include "Menu.h"
```
Comment (ZappoMan) [inline interface/src/Menu.cpp]: according the coding standard this should be "AppMenu.h"

**Comment by ZappoMan** [comment]
@jherico - "Looks like the depth reticle is only working when you're completely off the overlay," -- ahh that's interesting, that could be an issue with `getReticleOverDesktop()` -- Looks like maybe you had a merge conflict on `Q_PROPERTY(bool pointingAtSystemOverlay READ isPointingAtSystemOverlay)` of `ReticleInterface` The QML needs to call/set that to let the system know when it's pointing at the desktop or an overlay window... so that's likely the problem.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3321538b792e7ef4d22350d111d98b9d2547d79a available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp (hunk 5 above)
Comment (jherico) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: Yes, as soon as I work on that open issue this will be moved out, but it's a lot of code I want to preserve till I find the right place for it.

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp (hunk 6 above)
Comment (jherico) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: I dislike the creation of temp variables that get used once on the next one or two lines, but stay in scope for the remainder of the function.  It makes it harder to know what can be cleaned up since compilers are necessarily very bad at detecting unused variables that are not primitive types.

File: libraries/ui/src/ui/Menu.cpp (hunk 13 above)
Comment (jherico) [inline libraries/ui/src/ui/Menu.cpp]: removed

File: interface/src/Menu.h (hunk 14 above)
Comment (jherico) [inline interface/src/Menu.h]: Yes, but Menu.h is included in dozens of other places and I didn't want to have to go through and make a mess of all those unrelated files.

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 16]
```diff
@@ -0,0 +1,485 @@
...
+}
+
+
+glm::mat4 CompositorHelper::getReticleTransform(const glm::mat4& eyePose, const glm::mat4& headPose) const {
+    glm::mat4 result;
+    if (isHMD()) {
+        vec3 reticleScale = vec3(Cursor::Manager::instance().getScale() * reticleSize);
+        auto reticlePosition = getReticlePosition();
+        if (getReticleDepth() != 1.0f) {
+            auto spherical = overlayToSpherical(reticlePosition);
```
Comment (jherico) [inline libraries/display-plugins/src/display-plugins/CompositorHelper.cpp]: we can't use computeHmdPickRay here because it relies on the wrong camera, which is being updated asynchronously by the main thread.  Instead we need to use the head pose to compute the ray.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d4f636fc477b1fb829dde351197fbf454c0f337a available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7554b0c31bf86336e3e4085d60a1b72a8dc0ee68 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 36c4eb4559c435320bb63a85cd63546e8a0f4f68 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 267851faac1303c6910d6cfa9b1fc5d18d95ea88 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by cozza13** [comment]
I went through and tested this. Selecting on menus worked.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 05c007fe6ec1fdc92d6720945db7308155352cf7 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 984b6ce7fb712c3e35958c1dfbfae0f48de5f700 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2d01b88969e9ce0d5ae04f2ad96798ed042a746d available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3f0c2f8579fa865472101a5118b0f0d39c849eec available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8988dd9771eae5a5aeae6be0644aa080792d7bc5 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8988dd9771eae5a5aeae6be0644aa080792d7bc5 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8988dd9771eae5a5aeae6be0644aa080792d7bc5 available at: https://gist.github.com/2909bdddc3c7230fa095<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7221.dmg

**Comment by PhilipRosedale** [comment]
Is this now running fast?  Where are we on merging?

**Comment by ZappoMan** [comment]
It's still in testing.

**Comment by jherico** [comment]
@PhilipRosedale jitter in the overlays is in low-render-rate areas is gone... but in fact that was really the easy part and has been done for over a week.  The remaining time has been getting the mouse handling to work properly given the new division of code, where the compositor no longer has direct access to the Application type.  It was fiddly.  Should be complete now though.

**Comment by ZappoMan** [comment]
Testing report:
- Depth Reticle is now working again. - **Pass**
- Overlay is stready - **Pass**
- Loading/quick turn behavior seems the same as production - **Pass**
- Mouse dragging works - **Pass**
